diff --git a/source/gps_test.c b/source/gps_test.c
index 91f3791..874449d 100644
--- a/source/gps_test.c
+++ b/source/gps_test.c
@@ -81,7 +81,7 @@ int main(int argv, char **argc)
         /* Put this in a loop with a call to a high resolution sleep () in it. */
         if (gps_waiting(&GpsData, 500)) {
             //errno = 0;
-            if (gps_read(&GpsData) == -1) {
+            if (gps_read(&GpsData, NULL, 0) == -1) {
                 dprintf(1, "Read error\n");
                 //exit(1);
                 sleep(1);
